tests/test_aamp_ostinato.py

```diff
@@ -85,3 +85,60 @@ def test_deterministic_ostinatoed(seed, dask_cluster):
             npt.assert_almost_equal(ref_radius, comp_radius)
             npt.assert_almost_equal(ref_Ts_idx, comp_Ts_idx)
             npt.assert_almost_equal(ref_subseq_idx, comp_subseq_idx)
+
+
+def test_input_not_overwritten():
+    # aamp_ostinato preprocesses its input, a list of time series,
+    # by replacing nan value with 0 in each time series.
+    # This test ensures that the original input is not overwritten
+    m = 50
+    Ts = [np.random.rand(n) for n in [64, 128, 256]]
+
+    # insert at least one nan value
+    nan_size = [np.random.choice(np.arange(1, len(T) + 1)) for T in Ts]
```

Comment:
If you only need a single random integer for each time series then why not just do:

```
nan_size = [ np.random.randint(1, len(T) + 1) for T in Ts]
```

It doesn't make sense to generate a range of numbers just to extract a single value from it

Reply:
In fact, why should we allow so many elements to be `np.nan`? I'd limit it to as many as 10% of `len(T)` only

Reply:
> It doesn't make sense to generate a range of numbers just to extract a single value from it

Right! I do not know why my mind went to that complicated direction! I will revise it.

> In fact, why should we allow so many elements to be np.nan? I'd limit it to as many as 10% of len(T) only

Well, I think just a single `np.nan` should suffice actually. But then I thought why I should limit it to one element. So, I made it completely random.  I am just trying to understand "how" I should decide whether I should go with one random element, 10% of `len(T)` elements, or just any number of elements?

Reply:
Honestly, I would've gone with no np.nan but if you mist then maybe only do one. It's okay to hardcode the location too

Reply:
> I would've gone with no np.nan

I think we need at least one `np.nan` (in at least one of the time series in `Ts`) for this test function, as explained in [this comment](https://github.com/TDAmeritrade/stumpy/issues/982#issue-2337118449).  If there is no `np.nan`, then the following assertion always pass as the preprocessed data and the original data are the same.

```
npt.assert_almost_equal(T_ref[np.isfinite(T_ref)], T_comp[np.isfinite(T_comp)])
```

I will go with one `np.nan` and, I will hardcode the location (For the sake of simplicity, I can put it as the first element of each time series)

---

Note that in the other test function `test_extract_several_consensus`, there is no `np.nan` in the original input because `np.nan` will be created as we extract several consensus motifs.